src/engagement_db_coda_sync/configuration.py

```diff
@@ -1,37 +1,112 @@
+        :type update_users_and_code_schemes: bool
+        """
+        self.coda_dataset_id = coda_dataset_id
+        self.engagement_db_dataset = engagement_db_dataset
+        self.code_scheme_configurations = code_scheme_configurations
+        self.ws_code_match_value = ws_code_match_value
+        self.dataset_users_file_url = dataset_users_file_url
+        self.update_users_and_code_schemes = update_users_and_code_schemes
 
 
-@dataclass
 class CodaSyncConfiguration:
-    dataset_configurations: [CodaDatasetConfiguration]
-    ws_correct_dataset_code_scheme: CodeScheme
-    project_users_file_url: Optional[str] = None
-    set_dataset_from_ws_string_value: bool = False
-    default_ws_dataset: Optional[str] = None  # Engagement db dataset to move messages to if there is no dataset
-                                              # configuration for a particular ws_code_match_value. If None, crashes if
-                                              # a message is found with a WS label with a string value not in
-                                              # dataset_configurations. In most circumstances, this should be None as matching
-                                              # cases where there are no datasets usually indicates a missing piece of configuration.
+    def __init__(self, dataset_configurations, ws_correct_dataset_code_scheme, project_users_file_url=None,
+                 set_dataset_from_ws_string_value=False, default_ws_dataset=None):
+        """
+        Configuration for bidirectional sync between an engagement database and a Coda instance.
+
+        :param dataset_configurations: Configurations for each of the Coda datasets to sync.
+        :type dataset_configurations: list of CodaDatasetConfiguration
+        :param ws_correct_dataset_code_scheme: WS - Correct Dataset code scheme.
+                                               This will be added to every dataset in Coda, and allows messages that
+                                               have been assigned to the wrong dataset in Coda to be redirected to the
+                                               correct one. To configure this redirection, set the `ws_code_match_value`
+                                               properties in each `CodaDatasetConfiguration`.
+        :type ws_correct_dataset_code_scheme: core_data_modules.data_models.CodeScheme
+        :param project_users_file_url: GS URL to a json file containing a list of user ids, which each user id is a
+                                       string containing the email address of a user who should have permission to
+                                       access all the Coda datasets configured here.
+                                       If a dataset_configuration has its `dataset_users_file_url` property set,
+                                       the users will be updated from that file instead of the one referenced here.
```

Comment:
I suggest we have this at the end because args `ws_correct_dataset_code_scheme`,  `set_dataset_from_ws_string_value` and `default_ws_dataset` are related.

Reply:
good point, I have moved this as suggested